pkg/cli/clients/clients.go

```diff
@@ -137,40 +137,74 @@ type LogStream struct {
-	ListAllResourcesByApplication(ctx context.Context, applicationName string) ([]generated.GenericResource, error)
-	ListAllResourcesOfTypeInEnvironment(ctx context.Context, environmentName string, resourceType string) ([]generated.GenericResource, error)
-	ListAllResourcesByEnvironment(ctx context.Context, environmentName string) ([]generated.GenericResource, error)
-	ShowResource(ctx context.Context, resourceType string, resourceName string) (generated.GenericResource, error)
-	DeleteResource(ctx context.Context, resourceType string, resourceName string) (bool, error)
+	// ListResourcesOfType lists all resources of a given type in the configured scope.
+	ListResourcesOfType(ctx context.Context, resourceType string) ([]generated.GenericResource, error)
+
+	// ListResourcesOfTypeInApplication lists all resources of a given type in a given application in the configured scope.
+	ListResourcesOfTypeInApplication(ctx context.Context, applicationNameOrID string, resourceType string) ([]generated.GenericResource, error)
+
+	// ListResourcesOfTypeInEnvironment lists all resources of a given type in a given environment in the configured scope.
+	ListResourcesOfTypeInEnvironment(ctx context.Context, environmentNameOrID string, resourceType string) ([]generated.GenericResource, error)
+
+	// ListResourcesInApplication lists all resources in a given application in the configured scope.
+	ListResourcesInApplication(ctx context.Context, applicationNameOrID string) ([]generated.GenericResource, error)
+
+	// ListResourcesInEnvironment lists all resources in a given environment in the configured scope.
+	ListResourcesInEnvironment(ctx context.Context, environmentNameOrID string) ([]generated.GenericResource, error)
+
+	// GetResource retrieves a resource by its type and name (or id).
+	GetResource(ctx context.Context, resourceType string, resourceNameOrID string) (generated.GenericResource, error)
+
+	// DeleteResource deletes a resource by its type and name (or id).
+	DeleteResource(ctx context.Context, resourceType string, resourceNameOrID string) (bool, error)
+
+	// ListApplications lists all applications in the configured scope.
 	ListApplications(ctx context.Context) ([]corerp.ApplicationResource, error)
-	ShowApplication(ctx context.Context, applicationName string) (corerp.ApplicationResource, error)
-	GetGraph(ctx context.Context, applicationName string) (corerp.ApplicationGraphResponse, error)
 
-	// CreateOrUpdateApplication creates or updates an application.
-	CreateOrUpdateApplication(ctx context.Context, applicationName string, resource corerp.ApplicationResource) error
+	// GetApplication retrieves an application by its name (or id).
+	GetApplication(ctx context.Context, applicationNameOrID string) (corerp.ApplicationResource, error)
+
+	// GetApplicationGraph retrieves the application graph of an application by its name (or id).
+	GetApplicationGraph(ctx context.Context, applicationNameOrID string) (corerp.ApplicationGraphResponse, error)
```

Comment:
nice to see we can use ID or name everywhere.

Reply:
hmm, does this make -g flag somewhat redundant? that now would make sense only if the "name" was used, not id right?

Reply:
Not really. We'd still need to update the rest of the CLI to pass in a resource ID instead of a name. Right now the CLI only works with names and assumes the "default scope". 

So this change only really improves this specific layer.